### PR TITLE
Add lxqt-session, enable start in sddm

### PR DESCRIPTION
**Pull request by Besanon**

### CODE REVIEW

File: apparmor.d/groups/lxqt/lxqt-session [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  include <abstractions/qt5-shader-cache>
+  include <abstractions/nameservice-strict>
+
+  signal (send),
+  signal (receive) set=(kill, term) peer=startlxqt,
+  signal (receive) set=(kill, term) peer=sddm,
+
+  ptrace (read),
+
+  network netlink raw,
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/lxqt-session:24]: This rule should come before the signal rules.

File: apparmor.d/groups/lxqt/lxqt-session [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/4.0>,
+
+include <tunables/global>
+
+@{exec_path} = @{bin}/lxqt-session
+profile lxqt-session @{exec_path} flags=(attach_disconnected) {
+  include <abstractions/base>
+  include <abstractions/dbus-accessibility>
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/lxqt-session:13]: Use `bus-accessibility` instead (see https://apparmor.pujol.io/development/dbus/)

File: apparmor.d/groups/lxqt/lxqt-session [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  @{bin}/lxqt-leave                   rPx,
+  @{bin}/lxqt-about                   rPx,
+  @{bin}/lxqt-config-monitor          rPx,
+  @{bin}/dbus-update-activation-environment rCx -> dbus,
+  @{bin}/systemctl                    rCx -> systemctl,
+
+  @{bin}/pavucontrol                  rPx,
+  @{lib}/geoclue-2.0/demos/agent      rPx,
+  @{bin}/python3.@{int}               rPx,
+  @{lib}/python3.@{int}               rPx,
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/lxqt-session:51]: - There is no profile for `@{bin}/python3.@{int}`
- There is no such thing as an exec name: `@{lib}/python3.@{int}`

File: apparmor.d/groups/lxqt/lxqt-session [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+  /etc/xdg/                           r,
+  /etc/xdg/autostart/                 r,
+  /etc/xdg/autostart/*.desktop        r,
+  /etc/xdg/menus/lxqt-*               r,
+  /etc/xdg/openbox/*                  r,
+  /etc/udev/udev.conf                 r,
+
+  owner @{HOME}/.local/share/         r,
+  owner @{HOME}/.config/              r,
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/lxqt-session:77]: Remove, they are included in `lxqt`

File: apparmor.d/groups/lxqt/lxqt-session [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  owner @{HOME}/.config/autostart/    r,
+  owner @{HOME}/.config/autostart/*   rw,
+  owner @{user_cache_dirs}/openbox/   rw,
+  owner @{user_cache_dirs}/openbox/sessions/          rw,
+  owner @{user_cache_dirs}/openbox/openbox.log        rwk,
+  owner @{user_config_dirs}/mimeapps.list{,.@{rand6}} rw,
+  owner @{user_config_dirs}/dconf/user r,
+  owner @{user_config_dirs}/openbox/rc.xml            r,
+  owner @{user_share_dirs}/sddm/xorg-session.log      rw,
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/lxqt-session:86]: Remove, this is already part of `X-strict`

File: apparmor.d/groups/lxqt/lxqt-session [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+  @{PROC}/                             r,
+  @{PROC}/uptime                       r,
+  @{PROC}/@{pid}/stat                  r,
+  owner @{PROC}/@{pid}/stat            r,
+
+  @{run}/systemd/inhibit/**            rw,
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/lxqt-session:93]: That should be `@{att}/@{run}/systemd/inhibit/@{int}.ref rw,` and the rule should come before the `@{PROC}` ones.

File: apparmor.d/groups/lxqt/lxqt-session [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+  /dev/tty                             rw,
+
+  include if exists <local/lxqt-session>
+
+  profile systemctl {
+    include <abstractions/base>
+    include <abstractions/app/systemctl>
+
+    include if exists <local/lxqt-session_systemctl>
+
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/lxqt-session:104]: You forgot to close the systemctl subprofile: `}`

File: apparmor.d/groups/lxqt/lxqt-session [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  @{PROC}/                             r,
+  @{PROC}/uptime                       r,
+  @{PROC}/@{pid}/stat                  r,
+  owner @{PROC}/@{pid}/stat            r,
+
+  @{run}/systemd/inhibit/**            rw,
+
+  /dev/tty                             rw,
+
+  include if exists <local/lxqt-session>
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/lxqt-session:97]: Please more this include after the subprofile (it is where we place it for all other profiles).

**Comment by roddhjav** [comment]
That should be good as a first version of the profile. Note that this part of lxqt is most likelly an app launcher. If so, you can remove all `rPx` rule and add `include <abstractions/app-launcher-user>`

File: apparmor.d/groups/lxqt/lxqt-session (hunk 3 above)
Comment (Besanon) [reply apparmor.d/groups/lxqt/lxqt-session:51]: for now i will remove this, yes. I used a different profile for python. Only applicable with some backup programs

File: apparmor.d/groups/lxqt/lxqt-session (hunk 6 above)
Comment (Besanon) [reply apparmor.d/groups/lxqt/lxqt-session:93]: didnt reflected this change, going to check the other profiles as well

**Comment by Besanon** [comment]
which profile is supposed to be an app-launcher has been a bit confusing in the past. I will try to change this in lxqt-session, see if its ok to entirely remove the singel rPx-calls

**Comment by Besanon** [comment]
well, i added   
include <abstractions/app-launcher-user> 
it is now in lxqt-session and lxqt-panel. I am not sure about which programs still need to be called via rPx. There are some. Maybe not all are reflected in the changes. But it is working. Have a look

**Pull request merged**